src/configurationProvider.ts

```diff
@@ -134,7 +134,9 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                     }).sort ((a, b): number => {
                         return a.label > b.label ? 1 : -1;
                     });
-                    const selection = await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" });
+                    const selection = pickItems.length > 1 ?
+                        await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
+                        : pickItems[0];

```

Comment:
If there is no main class in the workspace, what happens here?

Reply:
early exit at the top of this method.

Reply:
Please extract the logic to a method, e.g. `promptForMainClass`